File: remove-comments/examples/python/calculator.py
#!/usr/bin/env python3
"""
A simple calculator module that demonstrates basic arithmetic operations.
This module provides functions for addition, subtraction, multiplication, and division.
"""

import math


def add(a, b):
    """
    Add two numbers and return the result.
    
    Args:
        a (float): First number
        b (float): Second number
        
    Returns:
        float: Sum of a and b
    """
    return a + b


def subtract(a, b):
    """
    Subtract second number from first number.
    
    Args:
        a (float): First number (minuend)
        b (float): Second number (subtrahend)
        
    Returns:
        float: Difference of a and b
    """
    return a - b


def multiply(a, b):
    """
    Multiply two numbers and return the result.
    
    Args:
        a (float): First number
        b (float): Second number
        
    Returns:
        float: Product of a and b
    """
    return a * b


def divide(a, b):
    """
    Divide first number by second number.
    
    Args:
        a (float): Dividend
        b (float): Divisor
        
    Returns:
        float: Quotient of a divided by b
        
    Raises:
        ValueError: If divisor is zero
    """
    if b == 0:
        raise ValueError("Cannot divide by zero")
    return a / b


def power(base, exponent):
    """
    Calculate base raised to the power of exponent.
    
    Args:
        base (float): Base number
        exponent (float): Exponent
        
    Returns:
        float: Result of base^exponent
    """
    return math.pow(base, exponent)


def main():
    """
    Main function to demonstrate calculator operations.
    """
    # Test basic arithmetic operations
    print("Calculator Demo")
    print("=" * 20)
    
    # Addition example
    x, y = 10, 5
    result = add(x, y)
    print("{} + {} = {}".format(x, y, result))
    
    # Subtraction example
    result = subtract(x, y)
    print("{} - {} = {}".format(x, y, result))
    
    # Multiplication example
    result = multiply(x, y)
    print("{} * {} = {}".format(x, y, result))
    
    # Division example
    try:
        result = divide(x, y)
        print("{} / {} = {}".format(x, y, result))
    except ValueError as e:
        print("Error: {}".format(e))
    
    # Power example
    result = power(x, y)
    print("{} ^ {} = {}".format(x, y, result))


# Execute main function if script is run directly
if __name__ == "__main__":
    main()

File: remove-comments/examples/c/calculator.c
/**
 * Simple Calculator in C
 * Author: Developer
 * Version: 1.0
 * 
 * This program demonstrates basic arithmetic operations
 */

#include <stdio.h>
#include <stdlib.h>

/* Define constants for operations */
#define MAX_INPUT 100
#define PI 3.14159

// Function prototypes
int add(int a, int b);       // Addition function
int subtract(int a, int b);  // Subtraction function
int multiply(int a, int b);  /* Multiplication function */
float divide(int a, int b);  /* Division function with float return */

/*
 * Main function - entry point of the program
 * Demonstrates basic calculator operations
 */
int main() {
    int num1, num2;
    char operation;
    
    // Print welcome message
    printf("Simple Calculator\n");
    printf("Enter first number: ");
    scanf("%d", &num1); // Read first number
    
    printf("Enter operation (+, -, *, /): ");
    scanf(" %c", &operation); // Note the space before %c
    
    printf("Enter second number: ");
    scanf("%d", &num2); // Read second number
    
    /* Process the operation using switch statement */
    switch(operation) {
        case '+':
            printf("Result: %d\n", add(num1, num2)); // Call add function
            break;
        case '-':
            printf("Result: %d\n", subtract(num1, num2)); // Call subtract
            break;
        case '*': /* Multiplication case */
            printf("Result: %d\n", multiply(num1, num2));
            break;
        case '/': /* Division case */
            if (num2 != 0) {
                printf("Result: %.2f\n", divide(num1, num2));
            } else {
                printf("Error: Division by zero!\n"); // Handle division by zero
            }
            break;
        default:
            printf("Error: Invalid operation!\n"); // Handle invalid input
            break;
    }
    
    return 0; // Success
}

/**
 * Add two integers
 * @param a First integer
 * @param b Second integer  
 * @return Sum of a and b
 */
int add(int a, int b) {
    return a + b; // Simple addition
}

/*
 * Subtract two integers
 */
int subtract(int a, int b) {
    int result = a - b; // Calculate difference
    return result; // Return the result
}

// Multiply two integers
int multiply(int a, int b) {
    /* Use a loop for demonstration */
    int result = 0;
    int i;
    
    // Add 'a' to result 'b' times
    for(i = 0; i < b; i++) {
        result += a; // Accumulate the sum
    }
    
    return result; // Return product
}

/**
 * Divide two integers and return float result
 */
float divide(int a, int b) {
    // Cast to float for accurate division
    return (float)a / (float)b; /* Return floating point result */
}